azurerm/resource_arm_application_insights_api_key.go

```diff
@@ -0,0 +1,179 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+
+	"github.com/Azure/azure-sdk-for-go/services/appinsights/mgmt/2015-05-01/insights"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmApplicationInsightsAPIKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmApplicationInsightsAPIKeyCreate,
+		Read:   resourceArmApplicationInsightsAPIKeyRead,
+		Delete: resourceArmApplicationInsightsAPIKeyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"api_key": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"application_insights_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"resource_group_name": resourceGroupNameSchema(),
+
+			"read_permissions": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Set:      schema.HashString,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validate.LowercaseString,
```

Comment:
Right now, the accepted values here would be `"agentconfig", "aggregate", "api", "draft", "extendqueries", "search"`, but the point of leaving this open to any input was to make it future proof in case Azure introduces new values.

Enforcing the currently known values will require to release a new version of the provider if that list changes. Is that really desired?

Reply:
I've  to implement this

Reply:
I've implemented this for now.